crates/holochain/src/core/workflow/sys_validation_workflow/validate_op_tests.rs

```diff
@@ -191,6 +191,81 @@ async fn validate_valid_agent_validation_package_op() {
     );
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn validate_delete_agent_key_op() {
+    holochain_trace::test_run();
+
+    let mut test_case = TestCase::new().await;
+
+    // Create agent pub key action
+    let create_agent_pub_key = Create {
+        author: test_case.agent.clone(),
+        action_seq: 2,
+        entry_type: EntryType::AgentPubKey,
+        entry_hash: test_case.agent.clone().into(),
+        prev_action: fixt!(ActionHash),
+        weight: Default::default(),
+        timestamp: Timestamp::now().into(),
+    };
+    let create_agent_pub_key_action = test_case
+        .sign_action(Action::Create(create_agent_pub_key))
+        .await;
+
+    // Op to validate
+    let mut action = fixt!(Delete);
+    action.author = test_case.agent.clone();
+    action.prev_action = create_agent_pub_key_action.as_hash().clone();
+    action.action_seq = create_agent_pub_key_action.action().action_seq() + 1;
+    action.deletes_entry_address = test_case.agent.clone().into();
+    action.timestamp = Timestamp::now();
+    let op = ChainOp::RegisterAgentActivity(fixt!(Signature), Action::Delete(action)).into();
```

Comment:
This is what I expected the code above to match. This test case makes sense to me and I think this is what we want to match the implementation to